{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "64fdc593",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pathlib\n",
    "import pandas as pd\n",
    "import seedir as sd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bc77417",
   "metadata": {},
   "source": [
    "### Parse Info\n",
    "This function parses sniffles column 8 (INFO) to extract the Type, End, Lenght, and Chr2 of the Strcutural variant. \n",
    "\n",
    "Sample data at index 7:\n",
    "```\n",
    "PRECISE;SVMETHOD=Snifflesv1.0.12;CHR2=1;END=3097326;ZMW=17;STD_quant_start=0.000000;STD_quant_stop=1.897367;Kurtosis_quant_start=2.000000;Kurtosis_quant_stop=1.946987;SVTYPE=INS;SUPTYPE=AL;SVLEN=3644;STRANDS=+-;STRANDS2=9,8,9,8;RE=17;REF_strand=14,14;Strandbias_pval=1;AF=0.377778\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "464dd0c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_info(line_array, infoIndex):\n",
    "    \n",
    "    dict_values={'SVTYPE':'NN', 'END':-1, 'CHR2':'', 'SVLEN':0}\n",
    "    \n",
    "    if(len(line_array)<=infoIndex):\n",
    "        return dict_values\n",
    "    \n",
    "    infoString=line_array[infoIndex]\n",
    "    info_array = infoString.split(\";\");\n",
    "    \n",
    "    if(len(info_array)<=2):\n",
    "        return dict_values\n",
    "    \n",
    "    for k, v in dict_values.items():\n",
    "        for item in info_array:\n",
    "            if(k+\"=\" in item):\n",
    "                dict_values[k]=item.replace(k+\"=\",\"\")\n",
    "                break\n",
    "    \n",
    "    return dict_values"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a05c16f",
   "metadata": {},
   "source": [
    "### Generate BED File\n",
    "\n",
    "The output file has the following format:\n",
    "\n",
    "Column | Data         | Comment                          \n",
    ":---   | ---          | ---                              \n",
    "Col1   | chr num      | 'chr' prefixed                   \n",
    "Col2   | start        | from sniffles output column (2)*  \n",
    "Col3   | end          | fom info data *                    \n",
    "Col4   | length       | fom info data              \n",
    "\n",
    "In case both start and end are the same (usually insertions), the start-end range will be modified by 20 bp at the start and end."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "36bc9935",
   "metadata": {},
   "outputs": [],
   "source": [
    "def bed_from_vcf(strain_name, input_file, final_dir, custom_mappings={}, min_len=-1, unified_file=False,\n",
    "                 split_precise=False, fix_coordinates=True, exclude_bnd=True, exlude_noend=True):\n",
    "\n",
    "    \"\"\"\n",
    "    bed_from_vcf(strain_name, input_file, final_dir, custom_mappings, unified_file, split_precise, \n",
    "                 fix_coordinates, exclude_bnd, exlude_noend)\n",
    "\n",
    "    Generates BED Files from VCF files (Currently supports Sniffles and PBSV).\n",
    "    \n",
    "    Parameters\n",
    "    ----------\n",
    "    strain_name : str\n",
    "        Name prefix for generated files\n",
    "    input_file : str\n",
    "        Input VCF\n",
    "    final_dir : str\n",
    "        Output Dir\n",
    "    custom_mappings : dict\n",
    "        Custom mapping for SV Type names\n",
    "    min_len: int\n",
    "        Minimum length for SVs (exclude SVs with length lower than). Set -1 to report all lengths.\n",
    "    unified_file: bool\n",
    "        A file is generated per SV Type unless this parameter is true.\n",
    "    split_precise, fix_coordinates, exclude_bnd, exlude_noend : bool\n",
    "        Whether or not the output will differentiate precise from non-precise SVs,\n",
    "        fix coordinates so start is always lower than end, and exclude entries \n",
    "        with SVTYPE=BND and no end position\n",
    "    \n",
    "        \n",
    "    Returns\n",
    "    -------\n",
    "    list\n",
    "        Merged list of SV Types found\n",
    "    \"\"\"\n",
    "    \n",
    "    #chr_names=[\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chrX\",\"chrY\"]\n",
    "    chr_names=[\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"X\"]\n",
    "    nucleotides = [\"A\", \"T\", \"G\", \"C\"]\n",
    "\n",
    "    type_dicts = {}\n",
    "    all_types = []\n",
    "    \n",
    "    with open(input_file, \"r\") as vcf_file:\n",
    "        for line in vcf_file:\n",
    "            line_array=line.strip().split(\"\\t\")\n",
    "\n",
    "            if(len(line_array) <8 or line.startswith(\"#\")):\n",
    "                continue\n",
    "\n",
    "            chr1=line_array[0]\n",
    "            start=int(line_array[1])-1\n",
    "            info=parse_info(line_array, 7)\n",
    "\n",
    "            end=int(info['END'])-1\n",
    "            length=abs(int(info['SVLEN']))\n",
    "            \n",
    "            if min_len > 0 and length < min_len :\n",
    "                continue\n",
    "\n",
    "            sv_type = info['SVTYPE'].replace(\"/\",\"\").upper()\n",
    "            sv_type = custom_mappings.get(sv_type, sv_type)\n",
    "            \n",
    "            if unified_file:\n",
    "                sv_type =  \"ALL\"\n",
    "\n",
    "            if((exclude_bnd and info['SVTYPE']=='BND') or (chr1 not in chr_names) or (exlude_noend and info['END']==-1)):\n",
    "                continue\n",
    "\n",
    "            if fix_coordinates and start > end:\n",
    "                aux = start\n",
    "                start = end\n",
    "                end = aux\n",
    "\n",
    "            if fix_coordinates and start == end:\n",
    "                start -= 20\n",
    "                end += 20\n",
    "\n",
    "            if split_precise:\n",
    "                if any(e in str(line_array[4]) for e in nucleotides):\n",
    "                    sv_type += \"_precise\"\n",
    "                else:\n",
    "                    sv_type += \"_other\"\n",
    "\n",
    "\n",
    "            type_entries = type_dicts.get(sv_type, [])\n",
    "            type_entries.append({\"chr\": chr1, \"start\": start, \"end\": end, \"len\": str(length)}) \n",
    "\n",
    "            type_dicts[sv_type]=type_entries\n",
    "        \n",
    "        pathlib.Path(final_dir).mkdir(parents=True, exist_ok=True)\n",
    "        \n",
    "        for type_dict in type_dicts.keys():\n",
    "                type_df = pd.DataFrame(type_dicts[type_dict])\n",
    "                type_df.to_csv(final_dir + type_dict + \"_\" + strain_name + \".bed\", sep='\\t', index=False, header=False)\n",
    "\n",
    "        all_types = list(set(all_types) | set([*type_dicts]))\n",
    "        \n",
    "    return all_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a767af79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['INV', 'DEL', 'DUP', 'INS']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sv_file_path = '../../data/DBA2J/calls/r64089e.minimap2_sniffles.vcf'\n",
    "# final_dir = '../../data/DBA2J/bed/50/minimap2-sniffles/'\n",
    "\n",
    "# sv_file_path = '../../data/DBA2J/calls/r64089e-ngmlr_sniffles.vcf'\n",
    "# final_dir = '../../data/DBA2J/bed/50/ngmlr-sniffles/'\n",
    " \n",
    "sv_file_path = '../../data/DBA2J/calls/r64089e-minimap2_pbsv.vcf'\n",
    "final_dir = '../../data/DBA2J/bed/50/minimap2-pbsv/'\n",
    "\n",
    "custom_mappings={\n",
    "    \"DEL\": \"DEL\",\n",
    "    \"DELINS\": \"DEL\",\n",
    "    \"DELLINKED\": \"DEL\",\n",
    "    \"INS\": \"INS\",\n",
    "    \"INSLINKED\": \"INS\",\n",
    "    \"INSLINKEDINS\": \"INS\",\n",
    "    \"INV\": \"INV\",\n",
    "    \"INVDELINS\": \"INV\",\n",
    "    \"INVINS\": \"INV\",\n",
    "    \"INVDUP\": \"INV\",\n",
    "    \"INVDEL\": \"INV\",\n",
    "    \"DELINV\": \"INV\",\n",
    "    \"DUP\": \"DUP\",\n",
    "    \"GAIN\": \"DUP\",\n",
    "    \"DUPINS\": \"DUP\",\n",
    "    \"CNV\": \"DUP\",\n",
    "    \"TANDEMDUP\": \"DUP\",\n",
    "    \"TANDEMDUPINV\": \"DUP\",\n",
    "    \"TANDEMLOWDUP\": \"DUP\",\n",
    "}\n",
    "\n",
    "bed_from_vcf(\"DBA2J\", sv_file_path, final_dir, custom_mappings, 50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "368386a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "📁 50/\n",
      "├─📁 intersects/\n",
      "│ ├─📄 DEL_catalog_vs_H1.bed\n",
      "│ ├─📄 DEL_catalog_vs_H2.bed\n",
      "│ ├─📄 DEL_catalog_vs_validated.bed\n",
      "│ ├─📄 DEL_pbsv_vs_H1.bed\n",
      "│ ├─📄 DEL_pbsv_vs_H2.bed\n",
      "│ ├─📄 DEL_pbsv_vs_catalog.bed\n",
      "│ ├─📄 DEL_pbsv_vs_validated.bed\n",
      "│ ├─📄 INS_catalog_vs_H6.bed\n",
      "│ ├─📄 INS_catalog_vs_H7.bed\n",
      "│ └─📄 INS_catalog_vs_validated.bed\n",
      "├─📁 minimap2-pbsv/\n",
      "│ ├─📄 DEL_DBA2J.bed\n",
      "│ ├─📄 DUP_DBA2J.bed\n",
      "│ ├─📄 INS_DBA2J.bed\n",
      "│ └─📄 INV_DBA2J.bed\n",
      "├─📁 minimap2-sniffles/\n",
      "│ ├─📄 DEL_DBA2J.bed\n",
      "│ ├─📄 DUP_DBA2J.bed\n",
      "│ ├─📄 INS_DBA2J.bed\n",
      "│ └─📄 INV_DBA2J.bed\n",
      "└─📁 ngmlr-sniffles/\n",
      "  ├─📄 DEL_DBA2J.bed\n",
      "  ├─📄 DUP_DBA2J.bed\n",
      "  ├─📄 INS_DBA2J.bed\n",
      "  └─📄 INV_DBA2J.bed\n"
     ]
    }
   ],
   "source": [
    "sd.seedir(\"../../data/DBA2J/bed/50/\", style='emoji', itemlimit=10, depthlimit=3, include_files='.*\\.bed$', regex=True, sort=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94cb67dc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:sv] *",
   "language": "python",
   "name": "conda-env-sv-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
